provisioning/WFLY-18671_Galleon_tooling_upward_compat.adoc

```diff
@@ -0,0 +1,219 @@
+---
+categories:
+  - provisioning
+  - galleon
+---
+= Upward compatibility for Galleon tooling
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Problem Overview
+
+Galleon core library that is used during provisioning is backward compatible. Galleon core version N is able to provision WildFly server that has been built using 
+Galleon core version N-x.
+
+Galleon core library is not upward compatible for good reason. It implements the provisioning logic for a given Galleon version. 
+Galleon core version N is not able (actually could but with no guarantee) to provision a WildFly server that depends on a Galleon core version N+x.
+
+The problem we are facing is that our provisioning tooling is bound to a given Galleon core version. 
+That makes upward compatibility impossible. Due to this limitation, each time we release new WildFly feature-packs that depend on a given Galleon core version, we need to upgrade all our tooling. Furthermore the error in this case is cryptic and doesn't help user to understand 
+what corrective action is required.
+
+This dependency of WildFly feature-pack on a Galleon core version is not formalized. It is only discovered at provisioning time, with cryptic errors.
+ 
+At feature-pack build time, even if the built feature-pack depends on a given Galleon core version, no check is operated. 
+Whatever the Galleon core in use, the galleon feature-pack is built without any compatibility check.
+
+We are here proposing to make WildFly provisioning tooling upward compatible and formalize the dependency that a feature-pack can have on a given Galleon version.
+
+== Proposal Overview
+
+=== New Galleon API
+
+In this proposal we are introducing a Galleon provisioning API that allows to deal with multiple versions of Galleon core library.
+
+The API isolates the tooling from the Galleon core version. It handles the Galleon feature-packs resolution and main provisioning use-cases.
+
+The API allows to provision feature-packs that depend on older Galleon core versions as well as newer Galleon core versions.
+
+=== Binding a feature-pack with a Galleon version at build time
+
+We have the opportunity to check that the Galleon core version used at build time will be able to provision the server at provisioning time.
+When building a feature-pack, the Galleon core version in use to build the feature-pack is set in the Galleon feature-pack `feature-pack.xml` descriptor.
+
+=== Impact on tooling
+
+This API is to be used by Provisioning tooling. For example:
+
+* Galleon CLI and Maven plugin
+* WildFly Maven plugin
+* WildFly bootable JAR Maven plugin
+* Prospero CLI
```

Comment:
just to clarify, this will affect other uses of Prospero not just CLI - management integration, installer, etc

Reply:
Thanks. fixed.